File: spring-boot-template/src/main/java/tui/meta/challenge/quotes/repository/QuoteRepository.java
package tui.meta.challenge.quotes.repository;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;
import tui.meta.challenge.quotes.repository.model.QuoteEntity;

import java.util.List;

public interface QuoteRepository extends MongoRepository<QuoteEntity, String> {

    @Query("{$text: { $search: ?0, $caseSensitive: false}}")
    List<QuoteEntity> getQuoteByAuthorFullTextSearch(String author);
}

File: micronaut-template/src/main/kotlin/tui/meta/challenge/repository/QuoteRepository.kt
package tui.meta.challenge.repository

import io.micronaut.data.mongodb.annotation.MongoRepository
import io.micronaut.data.repository.CrudRepository
import tui.meta.challenge.repository.model.QuoteEntity

@MongoRepository
interface QuoteRepository : CrudRepository<QuoteEntity, String> {
    fun findByQuoteAuthor(author: String): List<QuoteEntity>
}